testdata/sqllogictests/allowed_operations.slt

```diff
@@ -0,0 +1,14 @@
+statement ok
+CREATE EXTERNAL DATABASE debug_db FROM debug;
```

Comment:
this is cool, and I definitely 100% got to this point and don't fully understand how this works or would catch a bug if we accidentally regressed?

Reply:
We'll catch a bug here. Trying to insert should always check for permissions first. Moreover, we have an error message in the check that should force us to add different tests in the future if something regressed.